### DISCUSSION TITLE
Sample files nwipe.conf and nwipe_customers.csv

### DISCUSSION THREAD

**Discussion by tlschenkjr** [PartialVolume/shredos.x86_64#244] [General]
I'm working on updating the ShredOS images in our environment and see in the nwipe_launcher script code to use nwipe.conf and nwipe_customers.csv files.  I cannot however find any information on the format of these files.  It would be nice if the documentation was updated to explain the format of the files and provide examples of their usage.  At a minimum, samples of each would be greatly appreciated.

**Comment by PartialVolume** [comment]
Yes, I can add some info to the readme.

If /etc/nwipe/nwipe.conf and /etc/nwipe/nwipe_customers.csv do not exist when nwipe starts, they are automatically created and populated with a few values. In the case of the csv, the header is written:

Here's some examples:

nwipe_customers.csv (semi colon is used as the field delimiter, with each field in quotes)
```
"Customer Name";"Customer Address";"Contact Name";"Contact Phone"
"Test";"";"";""
"Canonical Group Limited";"The Office Group, St Dunstans House, 4th Floor, 201 Borough High St, SE1 1JA. London, UK.";"Mark Shuttleworth";"+44 20 7630 2400"
"Microsoft Corporation";"One Microsoft Way Redmond, WA 98052-6399 USA.";"Satya Nadella";"+001 (425) 882-8080"
"IBM Inc.";"1 New Orchard Rd, Armonk, New York, 10504, United States.";"Arvind Krishna";"+001 914-499-1900"
"Apple Inc.";"One Apple Park Way; Cupertino, CA 95014, U.S.A";"Tim Cook";"(800) APLCARE (8002752273)"
"Googleplex";"1600 Amphitheatre Parkway in Mountain View, California. U.S.A.";"Sundar Pichai";"+001 (650) 253-0000"
```

nwipe.conf (we use libconfig that handles the formatting of these files, but basically the config consists of key pairs contained within a group, so `Organisation_Details :` would be the group and the first key pair would be `Business_Name = "Not Applicable (BN)";`

```
Organisation_Details : 
{
  Business_Name = "Not Applicable (BN)";
  Business_Address = "Not Applicable (BA)";
  Contact_Name = "Not Applicable (BCN)";
  Contact_Phone = "Not Applicable (BCP)";
  Op_Tech_Name = "Not Applicable (OTN)";
};
PDF_Certificate : 
{
  PDF_Enable = "ENABLED";
  PDF_Preview = "DISABLED";
};
Selected_Customer : 
{
  Customer_Name = "Not Applicable (CN)";
  Customer_Address = "Not Applicable (CA)";
  Contact_Name = "Not Applicable (CCN)";
  Contact_Phone = "Not Applicable (CP)";
};
```

**Comment by tlschenkjr** [comment]
Sorry for the late reply, but I do appreciate the explanation above.  I got it working to my satisfaction and my customization efforts are going well.  Had to redo all of my sftp stuff that I added for the previous version.  Was also thinking it might be nice to write a python version of the nwipe_launcher script, but that's an "in my copious free time" project.

**Comment by morgandally** [comment]
I had this file before but now I can't find it. Where can i add the file to get the certificates working again?

**Reply by PartialVolume** [reply]
I'm not sure which file you mean exactly. Do you mean /etc/nwipe/nwipe.conf? You can edit it manually or just when nwipe starts and is showing the discs, press c which will display the configuration options, select the top option "Enable/Disable" hit return and it will toggle between enabled and disabled. To save to nwipe.conf just exit nwipe with Control C and the config file will get automatically updated.

![Screenshot_20240818_172535](https://github.com/user-attachments/assets/52506f6d-861e-4378-bc0c-94ad3c26c326)

**Reply by PartialVolume** [reply]
Just for reference nwipe.conf exists in multiple places, for instance it will exist on the USB boot disc and when ShredOS boots up, nwipe.conf (if it exists),  is read from the boot disc into RAM disc as /etc/nwipe/nwipe.conf. Nwipe uses the RAM disk version and if there are changes to it while nwipe is running they are written back to /etc/nwipe/nwipe.conf on the USB stick. This also applies if you configured ShredOS to operate with a ftp or lftp server.

**Reply by ehn29** [reply]
Hello

In case of using shredos on an lftp server and an ipxe boot, where should the nwipe.conf file be placed?
